### PR TITLE
Fix unnecessary hyphenation

### PR DESCRIPTION
**Pull request by wznmickey**
Fixes https://github.com/typst/typst/issues/5360
By manually set slight difference as 0.
The root reason is in grow in typst-layout/src/pad.rs, (s + p.abs) / (1.0 - p.rel.get()) is (31.592pt + 10.0pt)/(1.0-0.0). It should be 41.592pt but the given result is 41.59199999999999pt.

https://github.com/typst/typst/issues/5360#issuecomment-2465876365 but not sure how to solve this kind of calculation error.

### CODE REVIEW

File: crates/typst-layout/src/inline/linebreak.rs [hunk 1]
```diff
@@ -543,7 +543,14 @@ fn raw_ratio(
 ) -> f64 {
     // Determine how much the line's spaces would need to be stretched
     // to make it the desired width.
-    let delta = available_width - line_width;
+    let delta = {
+        let delta = available_width - line_width;
+        if delta.abs().to_pt() < 0.00001 {
```
Comment (laurmaedje) [inline crates/typst-layout/src/inline/linebreak.rs:548]: `delta.approx_eq(Abs::zero())` would be a little cleaner

Comment (wznmickey) [reply crates/typst-layout/src/inline/linebreak.rs]: Changed.

**Comment by wznmickey** [comment]
done

**Comment by laurmaedje** [comment]
I didn't catch this before, but I find the version in your first commit (with mutable let) a fair bit more readable actually.

**Comment by wznmickey** [comment]
Yes. But it will import a mutable `delta`.

**Comment by laurmaedje** [comment]
The function isn't that long, I don't think it matters much.

**Comment by wznmickey** [comment]
Changed to the first version.

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**